doc/src/advanced/custom_potential/external.md

```diff
@@ -0,0 +1,205 @@
+
+```rust
+pub trait PairPotential: Potential + BoxClonePair {
+    fn tail_energy(&self, cutoff: f64) -> f64;
+    fn tail_virial(&self, cutoff: f64) -> f64;
+
+    fn virial(&self, r: &Vector3D) -> Matrix3 { ... }
+}
+```
+
+First, we can see that `PairPotential` enforces the implementation of `Potential` which is denoted by `pub trait PairPotential: Potential ...` (we ignore `BoxClonePair` for now).
+That makes sense from a didactive point of view since we said that `PairPotential` is a "specialization" of `Potential`
+and furthermore, we can make use of all functions that we had to implement for `Potential`.
+
+There are three functions in the `PairPotential` trait. The first two functions start with `tail_`.
+These are functions to compute long range or tail corrections.
+Often, we introduce a cutoff distance into our potential beyond which we set the energy to zero.
+Doing so we intoduce an error which we can account for using a tail correction.
+We need two of these corrections, one for the energy, `tail_energy`, and one for the pressure (which uses `tail_virial`
+under the hood). For a beautiful derivation of tail corrections for truncated potentials, [see here](https://engineering.ucsb.edu/~shell/che210d/Simulations_of_bulk_phases.pdf).
+
+The third function, `virial`, already has its body implemented -- we don't have to write an implementation for
+our potential.
+
+We will omit the derivation of the formulae for tail corrections here but they are computed by solving these equations
+
+energy: $\int_{r_c}^{\infty} u(r) r^2 \mathrm{d}r$
+
+virial: $\int_{r_c}^{\infty} \frac{\partial u(r)}{\partial r} r^3 \mathrm{d}r$
+
+The implementation looks like so
+
+```rust
+impl PairPotential for Mie {
+    fn tail_energy(&self, cutoff: f64) -> f64 {
+        if self.m < 3.0 {
+            return 0.0;
+        };
+        let sigma_rc = self.sigma / cutoff;
+        let n_3 = self.n - 3.0;
+        let m_3 = self.m - 3.0;
+        let repulsive = f64::powf(sigma_rc, n_3);
+        let attractive = f64::powf(sigma_rc, m_3);
+        -self.prefac * self.sigma.powi(3) * (repulsive / n_3 - attractive / m_3)
+    }
+
+    fn tail_virial(&self, cutoff: f64) -> f64 {
+        if self.m < 3.0 {
+            return 0.0;
+        };
+        let sigma_rc = self.sigma / cutoff;
+        let n_3 = self.n - 3.0;
+        let m_3 = self.m - 3.0;
+        let repulsive = f64::powf(sigma_rc, n_3);
+        let attractive = f64::powf(sigma_rc, m_3);
+        -self.prefac * self.sigma.powi(3) * (repulsive * self.n / n_3 - attractive * self.m / m_3)
+    }
+}
+```
+
+Note that we cannot correct every kind of energy function.
+In fact, the potential has to be a *short ranged* potential.
+For our Mie potential, both the exponents have to be larger than 3.0 else our potential will be *long ranged* and
+the integral that has to be solved to compute the tail corrections diverges.
+For now, we will simply return zero in this case.
```

Comment:
The "for now" looks strange here: returning 0 is the right thing to do if the potential is not short-ranged